app/src/main/java/com/gathering/android/event/home/EventListViewModel.kt

```diff
@@ -66,9 +79,32 @@ class EventListViewModel @Inject constructor() : ViewModel() {
 
             }
         }
+
+        if (sortType == SortType.SORT_BY_DATE) {
+            filteredEventList = filteredEventList.sortedBy { it.date }
+        } else {
+            filteredEventList = filteredEventList.sortedWith(object : kotlin.Comparator<Event> {
+                override fun compare(o1: Event?, o2: Event?): Int {
+                    val distA = o1?.location?.distanceTo(getCurrentLocation())
+                    val distB = o2?.location?.distanceTo(getCurrentLocation())
+                    if (distA == null && distB == null) {
+                        return 0
+                    }
+                    if (distA == null) {
+                        return -1
+                    }
+                    if (distB == null) {
+                        return 1
+                    }
+                    return distA.compareTo(distB)
+                }
+
+            })
+        }
         return filteredEventList
     }
 
+
     private fun provideEventListMock(): List<Event> {
```

Comment:
lets take this method to a another class and call it `EventsRepository`. then pass a instance of that to this class as dependency. the class will later host the real code for calling actual API.